src/tools/tidy/src/lib.rs

```diff
@@ -124,6 +124,36 @@ pub fn git_diff<S: AsRef<OsStr>>(base_commit: &str, extra_arg: S) -> Option<Stri
     Some(String::from_utf8_lossy(&output.stdout).into())
 }
 
+/// Returns true if any modified file matches the predicate, if we are in CI, or if unable to list modified files.
+pub fn files_modified(ci_info: &CiInfo, pred: impl Fn(&str) -> bool) -> bool {
```

Comment:
This functions starts looking like
https://github.com/rust-lang/rust/blob/c720f49c46ec3e70bbc2bbb3e0dd1fe7fc0a47f2/src/build_helper/src/git.rs#L261

Maybe reuse it? And drop git_diff too.

Reply:
`git_diff` is used by several other parts of tidy tho, so getting rid of it would be much more work.

`git_git_modified_files` also returns a vec, which could be quite large in some cases, as well as using `diff-index`, which is a change in behavior vs what was done before.

this is already quite a large PR, I don't think it needs to also do general cleanup of `tidy` on top of that.

Reply:
Yeah, it seems to work for now, so let's keep it as it is.